rest/src/main/java/org/apache/unomi/rest/deserializers/ContextRequestDeserializer.java

```diff
@@ -0,0 +1,137 @@
+            if (requiredSessionPropertiesNode instanceof ArrayNode) {
+                List<String> requiredSessionProperties = new ArrayList<>();
+                requiredSessionPropertiesNode.elements().forEachRemaining(el -> requiredSessionProperties.add(el.textValue()));
+                cr.setRequiredProfileProperties(requiredSessionProperties);
+            }
+            if (node.get("requireScores") != null) {
+                cr.setRequireScores(node.get("requireScores").booleanValue());
+            }
+            final JsonNode eventsNode = node.get("events");
+            if (eventsNode instanceof ArrayNode) {
+                ArrayNode events = (ArrayNode) eventsNode;
+                List<Event> filteredEvents = new ArrayList<>();
+                for (JsonNode event : events) {
+                    if (schemaRegistry.isValid(event, "https://unomi.apache.org/schemas/json/events/" + event.get("eventType").textValue() + "/1-0-0")) {
+                        filteredEvents.add(jsonParser.getCodec().treeToValue(event, Event.class));
+                    }
+                }
+                cr.setEvents(filteredEvents);
+            }
+            final JsonNode filtersNode = node.get("filters");
+            if (filtersNode instanceof ArrayNode) {
+                ArrayNode filters = (ArrayNode) filtersNode;
+                List<PersonalizationService.PersonalizedContent> f = new ArrayList<>();
+                filters.elements().forEachRemaining(el -> {
+                    try {
+                        jsonParser.getCodec().treeToValue(el, PersonalizationService.PersonalizedContent.class);
+                    } catch (JsonProcessingException e) {
+                        // Unable to deserialize, ignore the entry
+                    }
+                });
+                cr.setFilters(f);
+            }
+            final JsonNode personalizationsNode = node.get("personalizations");
+            if (personalizationsNode instanceof ArrayNode) {
+                ArrayNode personalizations = (ArrayNode) personalizationsNode;
+                List<PersonalizationService.PersonalizationRequest> p = new ArrayList<>();
+                personalizations.elements().forEachRemaining(el -> {
+                    try {
+                        jsonParser.getCodec().treeToValue(el, PersonalizationService.PersonalizationRequest.class);
+                    } catch (JsonProcessingException e) {
+                        // Unable to deserialize, ignore the entry
+                    }
+                });
+                cr.setPersonalizations(p);
+            }
+            if (node.get("profileOverrides") != null) {
+                cr.setProfileOverrides(jsonParser.getCodec().treeToValue(node.get("profileOverrides"), Profile.class));
+            }
+            if (node.get("sessionPropertiesOverrides") != null) {
+                jsonParser.getCodec().treeToValue(node.get("sessionPropertiesOverrides"), Map.class);
```

Comment:
Also here it seems we are not assigning the value

Reply:
done